### PR TITLE
[sailfish-office] Add custom error UI for calligra documents. 

### PR DESCRIPTION
**Pull request by pvuorela**
Depends on:
https://git.merproject.org/mer-core/calligra/merge_requests/9

WIP at this point. Wanted to quickly try out at this point how my earlier error reporting gets to the app side. Already a lot better than the existing error reporting.

### CODE REVIEW

**Comment by pvuorela** [comment]
Adjusted layout a bit etc. Think good enough now.

File: plugin/FullscreenError.qml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        x: Theme.horizontalPageMargin
+        width: parent.width - 2*x
+        spacing: Theme.paddingMedium
+        anchors.verticalCenter: parent.verticalCenter
+
+        Image {
+            id: warningIcon
+
+            anchors.horizontalCenter: parent.horizontalCenter
+            source: "image://theme/icon-l-attention?" + Theme.highlightColor
```
Comment (adenexter) [inline plugin/FullscreenError.qml]: I'm not really sure where we stand going forward with the theme color provider, but maybe a HighlightImage and hightlighted: true instead here?

**Comment by adenexter** [comment]
Is the [sailfish-office] tag in the PR title without a bug number going to enrage CI?

**Comment by pvuorela** [comment]
Don't think PR title matters. Commit has a bug reference.

File: plugin/FullscreenError.qml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    id: root
+
+    property string error
+
+    anchors.fill: parent
+
+    Rectangle {
+        anchors.fill: parent
+        opacity: 0.4
+        color: Theme.highlightDimmerColor
```
[the remark this answers is not in the document]
Comment (pvuorela) [reply plugin/FullscreenError.qml]: Not sure, highlightBackgroundOpacity docs talk about usage with highlightBackgroundColor. For another thing isn't dimming kind of opposite to highlighting? In that sense the property name is strange?

File: plugin/FullscreenError.qml (hunk 1 above)
Comment (pvuorela) [reply plugin/FullscreenError.qml]: Sure.

**Pull request merged**